xdsl/transforms/convert_memref_to_ptr.py

```diff
@@ -0,0 +1,161 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from typing import cast
+
+from xdsl.context import MLContext
+from xdsl.dialects import arith, builtin, memref, ptr
+from xdsl.ir import Operation, SSAValue
+from xdsl.irdl import Any
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.exceptions import DiagnosticException
+
+
+# I think we also need to pass the adress width.
```

Comment:
Isn't it computed below?

Reply:
yeah, forgot to delete this comment from a previous version